packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,425 @@
+    final List<Widget> buttons = <Widget>[];
+    buttons.addAll(
+      children.map<Widget>(
+        (Widget child) {
+          return new ConstrainedBox(
+            constraints: const BoxConstraints(
+              minHeight: _kButtonHeight,
+              maxHeight: _kButtonHeight,
+            ),
+            child: new CustomPaint(
+              painter: new _CupertinoVerticalDividerPainter(),
+              child: child,
+            ),
+          );
+        },
+      ),
+    );
+
+    return new CupertinoScrollbar(
+      child: new SingleChildScrollView(
+        controller: actionScrollController,
+        child: new Column(
+          mainAxisSize: MainAxisSize.min,
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          children: buttons,
+        ),
+      ),
+    );
+  }
+}
+
+/// A button typically used in an [ActionSheet]. ADD MORE DOC HERE
+///
+/// See also:
+///
+///  * [ActionSheet], an alert that presents the user with a set of two or
+///    more choices related to the current context.
+class ActionSheetAction extends StatefulWidget {
+  ///Creates an action for an iOS-style action sheet. ADD MORE DOC HERE
+  const ActionSheetAction({
+    @required this.onPressed,
+    this.isDefaultAction = false,
+    this.isDestructiveAction = false,
+    @required this.child,
+  })  : assert(child != null),
+        assert(onPressed != null);
+
+  /// The callback that is called when the button is tapped.
+  ///
+  /// This attribute must not be null.
+  final VoidCallback onPressed;
+
+  /// Whether this action is the default choice in the action sheet.
+  ///
+  /// Default buttons have bold text.
+  final bool isDefaultAction;
+
+  /// Whether this action might change or delete data.
+  ///
+  /// Destructive buttons have red text.
+  final bool isDestructiveAction;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// Typically a [Text] widget.
+  final Widget child;
+
+  @override
+  _ActionSheetActionState createState() => _ActionSheetActionState();
+}
+
+class _ActionSheetActionState extends State<ActionSheetAction> {
+  Color _backgroundColor = CupertinoColors.white;
+
+  @override
+  Widget build(BuildContext context) {
+    TextStyle style = _kCupertinoActionSheetActionTextStyle;
+
+    if (widget.isDefaultAction) {
+      style = style.copyWith(fontWeight: FontWeight.w600);
+    }
+
+    if (widget.isDestructiveAction) {
+      style = style.copyWith(color: CupertinoColors.destructiveRed);
+    }
+
+    void _onTapDown(TapDownDetails event) {
+      setState(() {
+        _backgroundColor = _kSelectedColor;
+      });
+    }
+
+    void _onTapUp(TapUpDetails event) {
+      setState(() {
+        _backgroundColor = CupertinoColors.white;
+      });
+    }
+
+    void _onTapCancel() {
+      setState(() {
+        _backgroundColor = CupertinoColors.white;
+      });
+    }
+
+    void _onTap() {
+      widget.onPressed;
+      Navigator.pop(context); //SHOULD THE ACTION SHEET BE HANDLING THIS?
```

Comment:
No, the onPressed will handle that. Look at how the dialog demo in the gallery does it. In order to do it here, you'd have to know what value to pass back.